### PR TITLE
Adding the Stop Function

### PR DESCRIPTION
**Pull request by anagrath**
I was able to test on iOS, I dont have an android or windows phone, but assuming that the code came from the internet properly they should work as well.

### CODE REVIEW

**Comment by vilic** [comment]
Hmm, there are actually quite a lot problems in Java and C# code, see my inline comments in minutes.

File: src/ios/CDVTTS.m [hunk 1]
```diff
@@ -69,4 +69,7 @@ - (void)speak:(CDVInvokedUrlCommand*)command {
     [synthesizer speakUtterance:utterance];
 }
 
+- (void)stop:(CDVInvokedUrlCommand*)command {
+  [synthesizer stopSpeakingAtBoundary:AVSpeechBoundaryImmediate];
+}
```
Comment (vilic) [inline src/ios/CDVTTS.m]: indentation, 4 spaces.

File: src/wp/TTS.cs [hunk 2]
```diff
@@ -32,7 +32,9 @@ class TTS : BaseCommand {
         SpeechSynthesizer synth = new SpeechSynthesizer();
 
         string lastCallbackId;
-
+        public async void speak(string argsJSON) {
```
Comment (vilic) [inline src/wp/TTS.cs]: I guess you mean `stop`.

And you'll need `DispatchCommandResult` just like `speak` method.

File: www/tts.js [hunk 3]
```diff
@@ -25,4 +25,12 @@ exports.speak = function (text, onfulfilled, onrejected) {
         }, function (reason) {
             onrejected(reason);
         }, 'TTS', 'speak', [options]);
+};
+
+exports.stop = function (onfulfilled, onrejected) {
+    cordova.exec( function () {
```
Comment (vilic) [inline www/tts.js]: No space between `(` and `function`. And indentation, 4 spaces.

File: src/wp/TTS.cs [hunk 4]
```diff
@@ -32,7 +32,9 @@ class TTS : BaseCommand {
         SpeechSynthesizer synth = new SpeechSynthesizer();
 
         string lastCallbackId;
-
+        public async void speak(string argsJSON) {
+          synth.CancelAll();
```
Comment (vilic) [inline src/wp/TTS.cs]: Indentation, and please add empty lines between methods.

File: src/android/TTS.java [hunk 5]
```diff
@@ -90,6 +90,11 @@ public void onInit(int status) {
         }
     }
 
+    private void stop(JSONArray args, CallbackContext callbackContext)
+      throws JSONException, NullPointerException
+    {
+      tts.stop()
```
Comment (vilic) [inline src/android/TTS.java]: Indentation, 4 spaces. `{` starts at the end of previous line.
And, `tts.stop();`. `;` for Java is not optional.

**Comment by vilic** [comment]
And for Java version, please checkout https://github.com/anagrath/cordova-plugin-tts/blob/bd8b7a6cec2985b0a59a446ba692b2bf5682daba/src/android/TTS.java#L68 and add related branch to `stop`.

**Comment by anagrath** [comment]
Not sure what you mean by "add related branch to stop."  Happy to help though so just let me know.

**Comment by vilic** [comment]
Never mind, I will merge when I get time and update your code manually.

**Comment by anagrath** [comment]
Okay, I got you finally.  I added that to Android.  Please let me know.  My team is looking for the change.

**Comment by vilic** [comment]
@anagrath Oh you don't need to wait for me to merge the PR to use this plugin. Just install it directly from your fork. (Google "cordova install plugin from github" for more information.)

**Comment by vilic** [comment]
@anagrath Thanks!

**Comment by anagrath** [comment]
No problem. We may try to beef up the voice choices next. We r using the one voice but on a future sprint were thinking to make things more interesting with multiple voices at least for web  would be good if mobiles Match

**Pull request merged**